controllers/firewall_controller.go

```diff
@@ -435,24 +431,11 @@ func (r *FirewallReconciler) updateStatus(ctx context.Context, f firewallv1.Fire
 
 // SetupWithManager configures this controller to watch for the CRDs in a specific namespace
 func (r *FirewallReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	r.recorder = mgr.GetEventRecorderFor("FirewallController")
-	mapToFirewallReconcilation := handler.ToRequestsFunc(
-		func(a handler.MapObject) []reconcile.Request {
-			return []reconcile.Request{
-				{NamespacedName: types.NamespacedName{
-					Name:      firewallName,
-					Namespace: firewallNamespace,
-				}},
-			}
-		})
-	triggerFirewallReconcilation := &handler.EnqueueRequestsFromMapFunc{
-		ToRequests: mapToFirewallReconcilation,
-	}
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&firewallv1.Firewall{}).
 		// don't trigger a reconcilation for status updates
 		WithEventFilter(predicate.GenerationChangedPredicate{}).
-		Watches(&source.Kind{Type: &firewallv1.ClusterwideNetworkPolicy{}}, triggerFirewallReconcilation).
-		Watches(&source.Kind{Type: &corev1.Service{}}, triggerFirewallReconcilation).
+		For(&firewallv1.ClusterwideNetworkPolicy{}).
```

Comment:
I don't think this is correct thing to do. Comment to `For` [states](https://github.com/kubernetes-sigs/controller-runtime/blob/b704f447ea7c8f7059c6665143a4aa1f6da28328/pkg/builder/controller.go#L77):
> This is the equivalent of calling Watches(&source.Kind{Type: apiType}, &handler.EnqueueRequestForObject{})

So it will create request with CWNP name, but reconciler will try to load Firewall object with this name.

Reply:
Btw, that's one of the reasons why it would be good to add controller tests to firewall-controller. Should we create an issue for that?

Reply:
Thanks for the hint, i was also unsure, you can push your corrections to this branch.